src/storage/on_disk.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::Path;
+
+use alloy_primitives::{keccak256, Address, Bytes, FixedBytes, B256, B64, U256};
+use libmdbx::{Database, DatabaseOptions, NoWriteMap};
+use revm::primitives::Bytecode;
+
+use super::{AccountBasic, EvmCode, Storage};
+
+/// A storage that stores chain data in a MDBX database.
+#[derive(Debug)]
+pub struct OnDiskStorage {
+    inner: Database<NoWriteMap>,
+}
+
+impl OnDiskStorage {
+    /// Opens the on-disk storage at the specified path.
+    pub fn open(path: impl AsRef<Path>) -> Result<Self, libmdbx::Error> {
+        let db = Database::open_with_options(
+            &path,
+            DatabaseOptions {
+                max_tables: Some(16),
+                ..DatabaseOptions::default()
+            },
+        )?;
+        Ok(Self { inner: db })
+    }
+}
+
+impl Storage for OnDiskStorage {
+    type Error = libmdbx::Error;
+
+    fn basic(
+        &self,
+        address: &alloy_primitives::Address,
+    ) -> Result<Option<super::AccountBasic>, Self::Error> {
+        let tx = self.inner.begin_ro_txn()?;
```

Comment:
1. Can we reuse the same read-only transaction (pool) instead of creating a new one on each read?
2. Is it more performant to run concurrent transactions in separate threads for parallel mode?

Reply:
These are for performance optimization only, right? Can we postpone to another PR?

At least we should measure the current performance first. Otherwise, these optimizations look premature to me.

Reply:
We should benchmark and optimize right from this PR, so we don't have to check in and maintain unwanted code if it fails to demonstrate great speedup over in-memory storage. We can break the deduplicating bytecodes task down because each part is an instant improvement we want (to maintain).

These specific questions are important to improve our understanding. So that if on-disk storage shows promise but MDBX flops then hopefully we know why and seek a better alternative, like a more async I/O-friendly database and schema to avoid locks among worker threads.

So yeah, let's add a benchmark and optimize from there.